In [1]:
import pandas as pd

In [2]:
# Import the data
file = '../Resources/SFO_Airport_Utility_Consumption.csv'
utilities_df = pd.read_csv(file)

utilities_df.head()

Year  Month Number Month      Utility       Owner   Units       Usage
0  2013             1   Jan   Passengers      Campus     PAX   3209356.0
1  2013             1   Jan          Gas  Commission  Therms    363205.0
2  2013             1   Jan          Gas      Tenant  Therms     49393.0
3  2013             1   Jan  Electricity  Commission     kWh  12904353.0
4  2013             1   Jan  Electricity      Tenant     kWh  14002156.0

# Question: 
### Which utility's usage changed the most from 2013 to 2018?

In [3]:
# Data Collection and Cleaning
# Select only the needed columns


Year      Utility   Units       Usage
0  2013   Passengers     PAX   3209356.0
1  2013          Gas  Therms    363205.0
2  2013          Gas  Therms     49393.0
3  2013  Electricity     kWh  12904353.0
4  2013  Electricity     kWh  14002156.0

In [4]:
# Scale the Usage column to be more readable
# Rows with "Water" as the utility can be left as is


Year      Utility   Units      Usage
0  2013   Passengers     PAX   3.209356
1  2013          Gas  Therms   0.363205
2  2013          Gas  Therms   0.049393
3  2013  Electricity     kWh  12.904353
4  2013  Electricity     kWh  14.002156

In [5]:
# Alter the Units column to reflect the changes


Year      Utility           Units      Usage
0  2013   Passengers     Million PAX   3.209356
1  2013          Gas  Million Therms   0.363205
2  2013          Gas  Million Therms   0.049393
3  2013  Electricity     Million kWh  12.904353
4  2013  Electricity     Million kWh  14.002156

In [6]:
# Combine the Utility and Units columns
# by putting Units in parentheses


Year                    Utility           Units      Usage
0  2013   Passengers (Million PAX)     Million PAX   3.209356
1  2013       Gas (Million Therms)  Million Therms   0.363205
2  2013       Gas (Million Therms)  Million Therms   0.049393
3  2013  Electricity (Million kWh)     Million kWh  12.904353
4  2013  Electricity (Million kWh)     Million kWh  14.002156

In [7]:
# Create two new DataFrames with data from 2013
# and 2018 that each contain only the Utility and 
# Usage column. Reset the index  for each 
# DataFrame to Utility


Usage
Utility                             
Passengers (Million PAX)    4.191545
Gas (Million Therms)        0.273992
Gas (Million Therms)        0.057670
Electricity (Million kWh)  13.692172
Electricity (Million kWh)  12.342686

In [8]:
# Analyze

# Calculate the totals for each utility



{'Period': 2013, 'Passengers (Million PAX)': 45.00676399999999, 'Gas (Million Therms)': 3.88738209927, 'Electricity (Million kWh)': 330.284764, 'Water (Million Gallons)': 390.932476}
{'Period': 2018, 'Passengers (Million PAX)': 57.797231999999994, 'Gas (Million Therms)': 3.6217946029999997, 'Electricity (Million kWh)': 311.21944010000004, 'Water (Million Gallons)': 410.236112}


In [9]:
# Calculate the change per utility as a percentage
# of each utility's 2013 total.





# Create a dataframe with the results



Utility   2013   2018  Difference  Change %
0   Passengers (Million PAX)   45.0   57.8        12.8      28.4
1       Gas (Million Therms)    3.9    3.6        -0.3      -6.8
2  Electricity (Million kWh)  330.3  311.2       -19.1      -5.8
3    Water (Million Gallons)  390.9  410.2        19.3       4.9

In [10]:
# Set the index to the utility column

# Sort the rows based on Change %


2013   2018  Difference  Change %
Utility                                                      
Passengers (Million PAX)    45.0   57.8        12.8      28.4
Water (Million Gallons)    390.9  410.2        19.3       4.9
Electricity (Million kWh)  330.3  311.2       -19.1      -5.8
Gas (Million Therms)         3.9    3.6        -0.3      -6.8

# Summary
The number of passengers using the airport grew 28% in just 5 short years! Water consumption rose slightly at 4.9%, but despite the increase in airport traffic, Electricity and Gas usage were both down in 2018 compared to 2013, with Gas leading the charge with a 6.8% decline. 